### ISSUE TITLE
Filter Extension broken in 3.2.2

### ISSUE THREAD

**Issue by bendavies** [api-platform/core#5934]
**API Platform version(s) affected**: 3.2.2

**Description**  
ApiFilter annotations have stopped working in 3.2.2.

**How to reproduce**  
Given a. resource:

```php
#[ApiResource(
    shortName: 'MarketingCompany',
    operations: [
        new Get(),
        new GetCollection(),
    ],
    provider: EntityClassDtoStateProvider::class,
    stateOptions: new Options(entityClass: MarketingCompany::class),
)]
#[ApiFilter(SearchFilter::class, properties: [
    'name' => 'ipartial',
])]
class MarketingCompanyApi
{
    #[ApiProperty(readable: false, writable: false, identifier: true)]
    public ?int $id = null;
    public ?string $name;
}
```

Performing a request like:
```
GET /api/marketing-companies?name=foo
```

Will get to here:
https://github.com/api-platform/core/blob/ad2cbe073b1cf0d5589c2ef1e48603e4426286ad/src/Doctrine/Orm/Extension/FilterExtension.php#L54

the `$filterId` is:
`annotated_bind_hq_api_api_resource_marketing_company_api_api_platform_doctrine_orm_filter_search_filter`
but the service name in `ContainerInterface $filterLocator` is:
`annotated_bind_h_q_api_api_resource_marketing_company_api_api_platform_doctrine_orm_filter_search_filter`

the difference is: 
`annotated_bind_hq_` vs
`annotated_bind_h_q_`

**Possible Solution**  
I'm guessing this is down to https://github.com/api-platform/core/pull/5637

**Additional Context**

**Comment by soyuka** [comment]
Thanks for the report. Does `keep_legacy_inflector: true` solves the issue for now?

**Comment by bendavies** [comment]
> Thanks for the report. Does `keep_legacy_inflector: true` solves the issue for now?

yes, it does. 👍

**Comment by jotwea** [comment]
I am currently facing the same issue. Setting `keep_legacy_inflector: true` worked for me for the moment.

**Comment by gseric** [comment]
I think the issue is in the approach used for new Inflector implementation. It's singleton instead of proper service, so it's not configured properly at run time.

AP bundle extension configures it here: https://github.com/api-platform/core/blob/6d15e228611875a3c1a41a14ecf26a38bd67639a/src/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php#L986
by calling this static method: https://github.com/api-platform/core/blob/6d15e228611875a3c1a41a14ecf26a38bd67639a/src/Metadata/Util/Inflector.php#L35

However, that applies only at the container build time. During app run time, this static method is not called. Therefore, Inflector will always use its default config (Doctrine inflector) at run time.

So, in short:
1. service container filters IDs are generated using Symfony inflector (`BindHQ\Api\...` -> `annotated_bind_hq_api_...`) -- configurable
2. run time filters IDs are generated using Doctrine inflector (`BindHQ\Api\...` -> `annotated_bind_h_q_api_...`) -- not configurable

**Comment by soyuka** [comment]
feel free to provide a patch! Thanks

**Comment by soyuka** [comment]
After investigating this I think that we should definitely provide an inflector service instead of using a static inflector. It'll allow to override the inflector as well. I'm even thinking that we could revert https://github.com/api-platform/core/commit/6babb3d6b707290fdf314c0e96acd525d6f96670 and provide a better way upgrade path before 4.x.

**Comment by soyuka** [comment]
https://github.com/api-platform/core/pull/6447 fixes this I think

**Comment by VincentLanglet** [comment]
> [#6447](https://github.com/api-platform/core/pull/6447) fixes this I think

I think so cause `ApiFilter` worked fine on my side on 4.x so I think we can close this
